### ISSUE TITLE
escape $ character

### ISSUE THREAD

**Issue by ProjectPatatoe** [Icinga/icingaweb2-module-director#1617]
making a command that uses samba, pointing to a share of C$\blah. I can't input the $ and i also cant escape it as C\$\blah renders as C\\$\blah

## Your Environment

* Director version (System - About): 2075f2e54a38a05888a0dcc75fd69f3812b30c4e
* Icinga Web 2 version and modules (System - About): 2.6.1
* Icinga 2 version (`icinga2 --version`): r2.9.1-1
* Ubuntu 18.04.1 LTS
* apache2
* php 7.2.7-0ubuntu0.18.04.2

**Comment by dbu1986** [comment]
Have you tried escaping it with $?

```
C$$\blah
```

**Comment by ProjectPatatoe** [comment]
@dbu1986 It would seem that using C$$\blah appears as C$$\blah . The command receives C$$\blah as well

**Comment by lazyfrosch** [comment]
This should work, `$$` escapes the dollar sign from macro evaluation.

The backslash is escaped automatically by Director.

Input into field: `DOM$$\user:testpw`

```
object Service "http dummy login" {
    host_name = "dummytesthost"
    import "http"

    vars.http_auth_pair = "DOM$$\\user:testpw"
}
```

Executed Command:

```
'/usr/lib/nagios/plugins/check_http' '-H' 'dummytesthost' '-I' '127.0.0.1' '-a' 'DOM$\user:testpw' '-t' '10'
```

In what context are you setting the value?

**Comment by ProjectPatatoe** [comment]
Service -> Custom properties
I think I see what happened with my last comment. I saw that in the "preview" it showed as $$ so i thought the command would receive that as well. So yes, that does work, thanks!

I don't believe the icinga documentation states anywhere of a $$. The closest I've found https://www.icinga.com/docs/icinga2/latest/doc/17-language-reference/#string-literals-escape-sequences
